### PR TITLE
feat(tracing): Introduce function calls tracing [fixes LNG-169]

### PR DESCRIPTION
**Pull request by InversionSpaces**

### CODE REVIEW

File: api/aqua-api/.js/src/main/scala/api/types/InputTypes.scala [hunk 1]
```diff
@@ -45,6 +45,8 @@ class AquaConfig(
   @JSExport
   val noRelay: js.UndefOr[Boolean],
   @JSExport
+  val tracing: js.UndefOr[Boolean],
```
Comment (DieMyst) [inline api/aqua-api/.js/src/main/scala/api/types/InputTypes.scala:48]: you must also add this to `.d.ts` file in npm package

File: model/inline/src/test/scala/aqua/model/inline/ArrowInlinerSpec.scala [hunk 2]
```diff
@@ -309,6 +322,642 @@ class ArrowInlinerSpec extends AnyFlatSpec with Matchers {
...
+   *     retval = x + arg
+   *     <- retval
+   *   <- closure
+   *
+   * func outer() -> u16:
+   *   c <- inner(42)
+   *   retval = 37 + c(1) + c(2)
+   *   <- retval
+   */
+  "arrow inliner" should "leave meta after returned closue inlining" in {
```
Comment (DieMyst) [inline model/inline/src/test/scala/aqua/model/inline/ArrowInlinerSpec.scala:595]: typo `closue` -> `closure`

File: model/inline/src/test/scala/aqua/model/inline/ArrowInlinerSpec.scala [hunk 3]
```diff
@@ -399,18 +1049,22 @@ class ArrowInlinerSpec extends AnyFlatSpec with Matchers {
     model.equalsOrShowDiff(
       SeqModel.wrap(
         CallServiceModel(
-          LiteralModel("\"getSrv\"", LiteralType.string),
+          LiteralModel.liftString("getSrv"),
```
Comment (DieMyst) [inline model/inline/src/test/scala/aqua/model/inline/ArrowInlinerSpec.scala:1052]: we have `LiteralRaw.quote` maybe rename `liftString` to `quote` also?

File: model/src/main/scala/aqua/model/OpModel.scala [hunk 4]
```diff
@@ -142,9 +167,15 @@ case class CallServiceModel(serviceId: ValueModel, funcName: String, call: CallM
...
+
+  def apply(
+    serviceId: String,
+    funcName: String,
+    args: List[ValueModel],
+    result: VarModel
+  ): CallServiceModel =
     CallServiceModel(
-      LiteralModel(s"\"$serviceId\"", ScalarType.string),
+      LiteralModel.liftScalarString(serviceId),
```
Comment (DieMyst) [inline model/src/main/scala/aqua/model/OpModel.scala:178]: scalars cannot be literals. I think there can be some mess with variable names that, for some reason, were quoted

File: semantics/src/main/scala/aqua/semantics/Semantics.scala [hunk 5]
```diff
@@ -23,6 +23,8 @@ import cats.Reducible
 import cats.data.Validated.{Invalid, Valid}
 import cats.kernel.Monoid
 import cats.syntax.applicative.*
+import cats.syntax.option.*
+import cats.syntax.all.*
```
Comment (DieMyst) [inline semantics/src/main/scala/aqua/semantics/Semantics.scala:27]: don't use `syntax.all.*`

File: model/transform/src/main/scala/aqua/model/transform/funcop/Tracing.scala [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+      .getOrElse(
+        SeqModel.wrap(children.toList: _*)
+      )
+
+  override def folder: OpTransform.OpFolder = {
+    case (MetaModel.CallArrowModel(arrowName), children) =>
+      val child = getChild(children)
+
+      Eval.now(
+        enabledConfig
```
[the remark this answers is not in the document]
Comment (InversionSpaces) [reply model/transform/src/main/scala/aqua/model/transform/funcop/Tracing.scala:39]: Actually this code will just eliminate `MetaModel.CallArrowModel` from tree if `enabled == false`

[the remark this answers is not in the document]
Comment (DieMyst) [reply model/transform/src/main/scala/aqua/model/transform/funcop/Tracing.scala:39]: or, sounds legit

**Review by DieMyst — APPROVED** [review]

**Pull request merged**